### ISSUE TITLE
Feature Request: Add host user to environment variables

### ISSUE THREAD

**Issue by frob** [geerlingguy/drupal-vm#1958]
### Issue Type

 - Feature Idea

### Summary

I would find it useful to add the host username as a variable to the vagrant environment.

The specific issue I am trying to solve is passing the host user to the vm for use in drush scripts. This way I don't have to keep writing `drush --ssh-options="-l username"`. The agent is getting forwarded just fine so I ssh has access to my ssh-keys but I still have to set the username manually --which is a pain.

**Comment by phizev** [comment]
Have you tried setting the username in [example.drush.yml](https://github.com/drush-ops/drush/blob/9.x/examples/example.drush.yml) or [example.site.yml](https://github.com/drush-ops/drush/blob/9.x/examples/example.site.yml) (both are Drush 9 examples, Drush 8 doesn't use yml, but equivalent files exist.)? Or in [~/.ssh/config](https://www.ssh.com/ssh/config/)?

I have added post-provision tasks to copy such files into place.

**Comment by frob** [comment]
Right, either will work. This is actually how I am solving the problem, but it requires user action. 

The issue is that I don't want to have to have developer specific settings that the dev environment depends on and I work on a team. The post-provision tasks run as vagrant in the vm. I need a way to pass the host username over to them.

**Comment by phizev** [comment]
Anything dev specific in a project repo is no good, and from my view, there should be no user action past ```vagrant up``` to start using the VM for the first provision.

What I'm getting at is more having an ignored file which holds developer specific information.
As an example, we have a specific place to put the private SSH key so that it is imported to the VM, an SSH .config, or drush config, or site alias file could be done in the same manner. It requires a one off user action after cloning a repo, and before vagrant up, to edit a text file. Alternatively, set it in a variable in a config.local.yml, and use it from there if it's needed for Ansible tasks.

If you are in a position where the developers local username is always the same as the username they use on the remote sites, that functionality would be great, but I think it would still need a manual alternative if the usernames don't align. I'm also looking at this from the view point of a project with developers on Mac, Windows, and Linux, and I haven't had joy in SSH key forwarding from Windows.
